### DISCUSSION TITLE
disabledWhen & disabledUnless ideas

### DISCUSSION THREAD

**Discussion by mjamilasfihani** [tenancy/tenancy#310] [Ideas]
hi! I would like share some ideas about adding new method.

let's call it by : `disabledWhen` and `disabledUnless`

it's basically give some condition before executing : https://github.com/tenancy/hooks-migration/blob/47370400303fdd5f3831d6a2555d1a6d321c61f8/Events/ConfigureMigrations.php#L47 and also it can be implemented for the ConfigureSeeds.php too.

so the new method would be :

```php
public function disabledWhen(bool $status)
{
    if ($status) {
        return $this->disable();
    }

    return $this;
}

public function disabledUnless(bool $status)
{
    if (! $status) {
        return $this->disable();
    }

    return $this;
}
```

that's it, thank you.

**Comment by mjamilasfihani** [comment]
here is the detailed idea :

```php
<?php

declare(strict_types=1);

/*
 * This file is part of the tenancy/tenancy package.
 *
 * Copyright Tenancy for Laravel
 *
 * For the full copyright and license information, please view the LICENSE
 * file that was distributed with this source code.
 *
 * @see https://tenancy.dev
 * @see https://github.com/tenancy
 */

namespace Tenancy\Hooks\Migration\Events\Concerns;

trait Conditionable
{
    public function disabledWhen(bool $status): static
    {
        if ($status) {
            return $this->disable();
        }

        return $this;
    }

    public function disabledUnless(bool $status): static
    {
        if (! $status) {
            return $this->disable();
        }

        return $this;
    }
}
```

**Comment by ArlonAntonius** [comment]
Hi @mjamilasfihani,

I appreciate your suggestion and completely understand it, but what is the actual use case for us making it this way?

Our goal is to provide developers with easy access to the essential part for multi-tenancy in Laravel and we prefer a simple to-the-point API that allows them to do this. Adding more methods means more maintenance and a higher chance of adding breaking changes, so I would not directly like to add this unless there is a valid use case :)

**Reply by mjamilasfihani** [reply]
I got the point, thanks. hmm, so let's everything being "structured" already. since, yeah tenant is really "something" to implement. still, thank you mate 💯 @ArlonAntonius
